backend/src/interests/dto/interests.youtube.control.response.dto.ts

```diff
@@ -0,0 +1,30 @@
+import { interestsYoutubeControl } from '../definition/interests.youtube.control';
+
+export class InterestsYoutubeControlResponseDto {
+  requestType: string;
+  videoCurrentTime: number;
+  playStatus: string;
+  targetTime: number;
+  playSpeed: number;
+
+  constructor(
+    requestType: string,
+    videoCurrentTime: number,
+    playStatus: string,
+    targetTime: number,
+    playSpeed: number,
+  ) {
+    this.requestType = requestType;
+    this.videoCurrentTime = videoCurrentTime;
+    this.playStatus = playStatus;
+    this.targetTime = targetTime;
+    this.playSpeed = playSpeed;
+  }
+
+  static of(
+    requestType: interestsYoutubeControl,
+    options: { videoCurrentTime?: number; playStatus?: string; targetTime?: number; playSpeed?: number } = {},
+  ) {
+    return new this(requestType, options.videoCurrentTime, options.playStatus, options.targetTime, options.playSpeed);
+  }
```

Comment:
options 자체를 옵셔널로 구성하신 게 아닌, 빈 오브젝트로 디폴트 값을 두신 이유가 있으실까요?

Reply:
options 일부 속성만 넘겨주는 것 외에 아예 options 자체를 생략이 가능한 경우도 있어서 빈객체로 두어 options 에 접근할때 오류가 나지 않도록 했습니다.

Reply:
```suggestion
    options?: { videoCurrentTime?: number; playStatus?: string; targetTime?: number; playSpeed?: number },
  ) {
    return new this(requestType, options.videoCurrentTime, options.playStatus, options.targetTime, options.playSpeed);
  }
```
이렇게 options 자체를 옵셔널로 구성하는 것과 어떤 차이점이 있을까요?

Reply:
options 자체를 옵셔널로 하면 옵션이 없을때 options이 `undifined` 가 되어서 
`return new this(requestType, options.videoCurrentTime, options.playStatus, options.targetTime, options.playSpeed);`
이과정에서 에러가 나게 됩니다.

Reply:
만약 옵션을 하나도 넣지 않으면 constructor에 빈 값이 들어가서
videoCurrentTime, playStatus, targetTime, playSpeed에는 undefined가 들어가겠군요!

다만 해당 생성자 필드 속성에는
```
requestType: string;
videoCurrentTime: number;
playStatus: string;
targetTime: number;
playSpeed: number;
```
라고 되어 있어서 '해당 dto가 옵셔널 값을 지니고 있을 수 있는가?'가 표현이 잘 안될 수도 있을 것 같아요. 혹시 어떻게 생각하시나요?

Reply:
그것까진 생각을 못했네요 생성자 필드 속성과 생성자에서 옵셔널 속성을 부여하도록 수정하겠습니다.